packages/front-end/hooks/useSchemaFormOptions.ts

```diff
@@ -0,0 +1,56 @@
+import { useState } from "react";
+import {
+  InformationSchemaInterface,
+  InformationSchemaTablesInterface,
+} from "@/../back-end/src/types/Integration";
+import { DataSourceInterfaceWithParams } from "@/../back-end/types/datasource";
+import useApi from "./useApi";
+
+export default function useSchemaFormOptions(
+  datasource: DataSourceInterfaceWithParams
```

Comment:
I believe this is actually nullable.  You can open the MetricForm even if you don't have a data source added to your account.